modules/base/src/processors/volumevoronoisegmentation.cpp

```diff
@@ -0,0 +1,104 @@
+
+#include <modules/base/processors/volumevoronoisegmentation.h>
+#include <modules/base/algorithm/volume/volumevoronoi.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VolumeVoronoiSegmentation::processorInfo_{
+    "org.inviwo.VolumeVoronoiSegmentation",  // Class identifier
+    "Volume Voronoi Segmentation",           // Display name
+    "Undefined",                             // Category
+    CodeState::Experimental,                 // Code state
+    Tags::None,                              // Tags
+};
+const ProcessorInfo VolumeVoronoiSegmentation::getProcessorInfo() const { return processorInfo_; }
+
+VolumeVoronoiSegmentation::VolumeVoronoiSegmentation()
+    : Processor(), volume_("inputVolume"), inputMesh_("inputPositions"), outport_("outport") {
+
+    addPort(volume_);
+    addPort(inputMesh_);
+    addPort(outport_);
+}
+
+void VolumeVoronoiSegmentation::process() {
+    auto inputMesh = inputMesh_.getData();
+
+    // Get seed points
+    // TODO: see if this can be done in a better way
+    auto posIt = util::find_if(inputMesh->getBuffers(), [](const auto& buf) {
+        return buf.first.type == BufferType::PositionAttrib;
+    });
+    if (posIt == inputMesh->getBuffers().end()) {
+        return;
+    }
+    const auto positionRam = posIt->second->getRepresentation<BufferRAM>();
+    std::vector<dvec3> positions;
+    for (std::size_t i = 0; i < positionRam->getSize(); ++i) {
+        positions.push_back(positionRam->getAsDVec3(i));
+    }
+
+    // Get indices for seed points (if exists??)
+    // TODO: should probably have some fallback solution if there are no indices...?
+    auto indexIt = util::find_if(inputMesh->getBuffers(), [](const auto& buf) {
+        return buf.first.type == BufferType::IndexAttrib;
+    });
+    if (indexIt == inputMesh->getBuffers().end()) {
+        return;
+    }
+    // TODO: see if this can be done in a better way
+    const auto& indices = static_cast<const BufferRAMPrecision<uint32_t, BufferTarget::Data>*>(
+                              indexIt->second->getRepresentation<BufferRAM>())
+                              ->getDataContainer();
+
+    if (positions.size() != indices.size()) {
+        return;
+    }
+
+    // Assuming the positions and indices come in the same order...
+    std::unordered_map<dvec3, uint32_t> seedPointsWithIndices;
+    for (std::size_t i = 0; i < positions.size(); ++i) {
+        seedPointsWithIndices.emplace(positions[i], indices[i]);
+    }
+
+    // In the case of the cube data volume,
+    // the voxel positions needs to be divided with the dimensions of the volume
+    const auto volume = volume_.getData();
+    const auto basis = volume->getBasis();
+    const auto dimensions = volume->getDimensions();
+    const auto scaledBasis =
+        mat3{basis[0] / dimensions.x, basis[1] / dimensions.y, basis[2] / dimensions.z};
```

Comment:
see previous comments about the coordinate transformers.

Reply:
Yep I hope I have fixed this now to be correct!